### PR TITLE
WIP: Feature/cart item stock check

### PR DESCRIPTION
**Pull request by ChristopherSchwab**

### CODE REVIEW

File: category/domain/category.go [hunk 1]
```diff
@@ -25,10 +25,18 @@ type (
...
 		CategoryTypeCode   string
 		CategoryAttributes Attributes
+		Promotion          Promotion
 	}
 
 	// Attributes define additional category attributes
 	Attributes map[string]interface{}
+
+	// Promotion defines promotion for a category
+	Promotion struct {
```
Comment (danielpoe) [inline category/domain/category.go]: Why is this part of this PR?

File: cart/application/cartService.go [hunk 2]
```diff
@@ -861,3 +861,33 @@ func (cs *CartService) dispatchAllEvents(ctx context.Context, events []flamingo.
 		cs.eventRouter.Dispatch(ctx, e)
 	}
 }
+
+// CheckAndAdjustStockInCart - checks if the stock of an item in the cart is lower than the quantity of it and if so updates the quantity
+func (cs *CartService) CheckAndAdjustStockInCart(ctx context.Context, session *web.Session) error {
```
Comment (danielpoe) [inline cart/application/cartService.go]: Stock is a "concept" that was not known to the cart.
The cart has the concept of "QtyRestrictors"  and we should use this as naming.
e.g. AdjustToRestrictedQty

The result of that function should be a value object that descibes what has been adjusted (e.g. slice of AdjustmentInfo (cartid, old, new qty, restrictor?)

File: cart/interfaces/controller/cartcontroller.go [hunk 3]
```diff
@@ -161,6 +167,11 @@ func (cc *CartViewController) UpdateQtyAndViewAction(ctx context.Context, r *web
 		}
 	}
 
+	err = cc.applicationCartService.CheckAndAdjustStockInCart(ctx, r.Session())
```
Comment (danielpoe) [inline cart/interfaces/controller/cartcontroller.go]: Imho we should enable this via feature flag - and per default it should be deactivated

File: cart/interfaces/controller/cartcontroller.go [hunk 4]
```diff
@@ -126,6 +126,12 @@ func (cc *CartViewController) AddAndViewAction(ctx context.Context, r *web.Reque
 		return cc.responder.Render("checkout/carterror", nil)
 	}
 
+	err = cc.applicationCartService.CheckAndAdjustStockInCart(ctx, r.Session())
```
Comment (danielpoe) [inline cart/interfaces/controller/cartcontroller.go]: use featureflag here as well

File: category/domain/category.go (hunk 1 above)
Comment (ChristopherSchwab) [reply category/domain/category.go]: There was a mistake while rebasing. Will close this merge request and open a new one. Will also copy over your other comments.

**Pull request closed without merging**